src/run.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as logging from 'plylog';
+import * as resolve from 'resolve';
+import * as updateNotifier from 'update-notifier';
+import * as cliTypeOnly from './polymer-cli';
+
+const packageJson = require('../package.json');
+
+const logger = logging.getLogger('cli.main');
+
+
+// Update Notifier: Asynchronously check for package updates and, if needed,
+// notify on the next time the CLI is run.
+// See https://github.com/yeoman/update-notifier#how for how this works.
+updateNotifier({pkg: packageJson}).notify();
+
+resolve('polymer-cli', {basedir: process.cwd()}, async(_error, path) => {
```

Comment:
The error here is the most common case, when the package didn't resolve. It's handled on the next line's check for `path`.